File: Brian Kernighans algorithm/count_set_bits_brute_force.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        while n:
            count += n & 1 # check last bit
            n = n >> 1
            
        return count

if __name__ == "__main__":
    solution = Solution()
    number = 22
    print("{0:d} in binary is {1:b}".format(number, number))
    print("the number of set bits in {0:d} is {1:d}".format(number, solution.hammingWeight(number)))
    
"""
22 in binary is 10110
the number of set bits in 22 is 3
"""

File: Brian Kernighans algorithm/count_set_bits_brute_force.go
package main

import "fmt"

func hammingWeight(num uint32) int {
	var count uint32
	for num > 0 {
		count += num & 1 // check last bit
		num = num >> 1
	}

	return int(count)
}

func main() {
	number := 22
	fmt.Printf("%d in binary is %b\n", number, number)
	fmt.Printf("the number of set bits in %d is %d\n", number, hammingWeight(22))
}

/*
output:
22 in binary is 10110
the number of set bits in 22 is 3
*/
